### PR TITLE
lcov update

### PR DESCRIPTION
**Pull request by billschereriii**

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -108,6 +108,25 @@ jobs:
           echo "CXX=icpx" >> $GITHUB_ENV &&
           echo "FC=ifort" >> $GITHUB_ENV
 
+      # Set up perl environment for LCOV
+      - uses: actions/checkout@v3
+      - name: Setup perl
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: '5.30'
+          install-modules: Memory::Process
```
Comment (ashao) [inline .github/workflows/run_tests.yml:117]: Any particular reason why this couldn't be installed with apt?

File: Makefile [hunk 2]
```diff
@@ -596,12 +596,12 @@ third-party/catch/single_include/catch2/catch.hpp:
...
-third-party/lcov/install/usr/local/bin/lcov:
+lcov: third-party/lcov/install/bin/lcov
+third-party/lcov/install/bin/lcov:
 	@mkdir -p third-party
 	@cd third-party && \
 	git clone $(LCOV_URL) lcov --branch $(LCOV_VER) --depth=1
 	@cd third-party/lcov && \
 	mkdir -p install && \
-	CC=gcc CXX=g++ DESTDIR="install/" make install && \
+	CC=gcc CXX=g++ make PREFIX=`pwd`/install/ install && \
```
Comment (ashao) [inline Makefile:606]: I'd slightly prefer `$PWD` or even `$(pwd)`

**Review by ashao — APPROVED** [review]
Very minor comments. Otherwise it looks good to me!

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (billschereriii) [reply .github/workflows/run_tests.yml:117]: library not found, so we had to do it this way

File: Makefile (hunk 2 above)
Comment (billschereriii) [reply Makefile:606]: No objections to that. I just went with what I knew :)

Comment (billschereriii) [reply Makefile:607]: Oddly, $(PWD) and $(CURDIR) are both failing

**Pull request merged**